gensim/test/test_keras_integration.py

```diff
@@ -0,0 +1,147 @@
+import unittest
+import os
+import sys
+import keras
+import numpy as np
+from gensim.models import word2vec
+from keras.engine import Input
+from keras.models import Model
+from keras.layers import merge
+from keras.preprocessing.text import Tokenizer
+from keras.preprocessing.sequence import pad_sequences
+from keras.utils.np_utils import to_categorical
+from keras.layers import Dense, Flatten
+from keras.layers import Conv1D, MaxPooling1D
+
+sentences = [
+    ['human', 'interface', 'computer'],
+    ['survey', 'user', 'computer', 'system', 'response', 'time'],
+    ['eps', 'user', 'interface', 'system'],
+    ['system', 'human', 'system', 'eps'],
+    ['user', 'response', 'time'],
+    ['trees'],
+    ['graph', 'trees'],
+    ['graph', 'minors', 'trees'],
+    ['graph', 'minors', 'survey']
+]
+
+module_path = os.path.dirname(__file__)  # needed because sample data files are located in the same folder
+datapath = lambda fname: os.path.join(module_path, 'test_data', fname)
+
+
+class TestKerasWord2VecWrapper(unittest.TestCase):
+    def setUp(self):
+        self.model_cos_sim = word2vec.Word2Vec(sentences, size=100, min_count=1, hs=1)
+        self.model_twenty_ng = word2vec.Word2Vec(word2vec.LineSentence(datapath('20_newsgroup_keras_w2v_data.txt')), min_count=1)
+
+    def testWord2VecTraining(self):
+        """
+        Test word2vec training.
+        """
+        model = self.model_cos_sim
+        self.assertTrue(model.wv.syn0.shape == (len(model.wv.vocab), 100))
+        self.assertTrue(model.syn1.shape == (len(model.wv.vocab), 100))
+        sims = model.most_similar('graph', topn=10)
+        # self.assertTrue(sims[0][0] == 'trees', sims)  # most similar
+
+        # test querying for "most similar" by vector
+        graph_vector = model.wv.syn0norm[model.wv.vocab['graph'].index]
+        sims2 = model.most_similar(positive=[graph_vector], topn=11)
+        sims2 = [(w, sim) for w, sim in sims2 if w != 'graph']  # ignore 'graph' itself
+        self.assertEqual(sims, sims2)
+
+    def testEmbeddingLayerCosineSim(self):
+        """
+        Test Keras 'Embedding' layer returned by 'get_embedding_layer' function for a simple word similarity task.
```

Comment:
@tmylk Since we are not using any new extra array/matrix in `get_embedding_layer()`, no extra memory is being used because of the function. Thus, as expected, memory profiler shows no extra memory usage because of the function.